test/Unosquare.Swan.Test/ObjectComparerTest.cs

```diff
@@ -11,76 +9,170 @@ namespace Unosquare.Swan.Test
     [TestFixture]
     public class ObjectComparerTest
     {
-        [Test]
-        public void CompareEqualsObjectsTest()
-        {
-            var left = DateBasicJson.GetDateDefault();
-            var right = DateBasicJson.GetDateDefault();
-
-            Assert.IsTrue(ObjectComparer.AreObjectsEqual(left, right));
-        }
-
-        [Test]
-        public void CompareDifferentObjectsTest()
-        {
-            var left = BasicJson.GetDefault();
-            var right = new BasicJson();
-
-            Assert.IsFalse(ObjectComparer.AreObjectsEqual(left, right));
-        }
 
-        [Test]
-        public void CompareEqualsStructsTest()
+        [TestFixture]
+        public class AreObjectsEqual : ObjectComparerTest
         {
-            var left = new SampleStruct();
-            var right = new SampleStruct();
-
-            Assert.IsTrue(ObjectComparer.AreStructsEqual(left, right));
+            [Test]
+            public void CompareEqualsObjects_ReturnsTrue()
+            {
+                var left = DateBasicJson.GetDateDefault();
+                var right = DateBasicJson.GetDateDefault();
+
+                Assert.IsTrue(ObjectComparer.AreObjectsEqual(left, right));
+            }
+
+            [Test]
+            public void CompareDifferentObjects_ReturnsFalse()
+            {
+                var left = BasicJson.GetDefault();
+                var right = new BasicJson();
+
+                Assert.IsFalse(ObjectComparer.AreObjectsEqual(left, right));
+            }
+
+            [Test]
+            public void WithDifferentObjects_ReturnsFalse()
```

Comment:
WithDifferentObjects and CompareDifferentObjects (line 26)...what's the difference?